python/tests/test_ownership_variable.py

```diff
@@ -0,0 +1,543 @@
+    outer.value['x', 0] = sc.scalar('asd')
+    inner['x', 1] = sc.scalar('qwe')
+    assert sc.identical(
+        outer, make_variable(make_variable(np.array(['asd', 'qwe', 'mno']))))
+    assert sc.identical(inner, make_variable(np.array(['asd', 'qwe', 'mno'])))
+    # TODO `a` seems to be invalidated
+    # np.testing.assert_array_equal(a, ['abc', 'def'])
+
+
+def test_own_var_1d_set():
+    # Input arrays are not shared.
+    a = np.arange(5)
+    v = make_variable(a)
+    v['x', 0] = -1
+    v.values[1] = -2
+    a[2] = -3
+    v['x', 3:] = [-4, -5]
+    assert sc.identical(v, make_variable([-1, -2, 2, -4, -5]))
+    np.testing.assert_array_equal(a, np.array([0, 1, -3, 3, 4]))
+
+
+def test_own_var_1d_get():
+    # .values getter shares ownership of the array.
+    v = make_variable(np.arange(5))
+    a = v.values
+    v['x', 0] = -1
+    v.values[1] = -2
+    a[2] = -3
+    v['x', 3:] = [-4, -5]
+    expected = [-1, -2, -3, -4, -5]
+    assert sc.identical(v, make_variable(expected))
+    np.testing.assert_array_equal(a, expected)
+
+
+def test_own_var_1d_copy():
+    # Depth of copies of variables can be controlled.
+    v = make_variable(np.arange(5.0), unit='m')
+    v_copy = copy(v)
+    v_deepcopy = deepcopy(v)
+    v_methcopy = v.copy(deep=False)
+    v_methdeepcopy = v.copy(deep=True)
+
+    v.values = np.arange(5.0, 10)
+    v['x', 0] = -1.0
+    v.unit = 's'
+    modified = make_variable([-1.0, 6.0, 7.0, 8.0, 9.0], unit='s')
+    original = make_variable([0.0, 1.0, 2.0, 3.0, 4.0], unit='m')
+    assert sc.identical(v, modified)
+    assert sc.identical(v_copy, modified)
+    assert sc.identical(v_deepcopy, original)
+    assert sc.identical(v_methcopy, modified)
+    assert sc.identical(v_methdeepcopy, original)
+
+    v.variances = np.arange(10.0, 15.0)
+    np.testing.assert_array_equal(v_copy.variances, v.variances)
+    assert v_deepcopy.variances is None
+    np.testing.assert_array_equal(v_methcopy.variances, v.variances)
+    assert v_methdeepcopy.variances is None
+
+    v_copy = copy(v)
+    v_deepcopy = deepcopy(v)
+    v_methcopy = v.copy(deep=False)
+    v_methdeepcopy = v.copy(deep=True)
+    v.values[1] = -2.0
+    v.variances[1] = -12.0
+    v.unit = 'kg'
+    original = make_variable([-1.0, 6.0, 7.0, 8.0, 9.0],
+                             variances=[10.0, 11.0, 12.0, 13.0, 14.0],
+                             unit='s')
+    modified = make_variable([-1.0, -2.0, 7.0, 8.0, 9.0],
+                             variances=[10.0, -12.0, 12.0, 13.0, 14.0],
+                             unit='kg')
+    assert sc.identical(v, modified)
+    assert sc.identical(v_copy, modified)
+    assert sc.identical(v_deepcopy, original)
+    assert sc.identical(v_methcopy, modified)
+    assert sc.identical(v_methdeepcopy, original)
+
+
+def test_own_var_1d_pyobj_set():
+    # Input data is deep-copied.
+    x = {'num': 1, 'list': [2, 3]}
+    y = {'num': 4, 'list': [5, 6]}
```

Comment:
As above, we should probably change this? It made sense to deep-copy under our old mechanism, but now not any more.